synapse/handlers/room_member.py

```diff
@@ -701,7 +702,36 @@ async def update_membership_locked(
                     # so don't really fit into the general auth process.
                     raise AuthError(403, "Guest access not allowed")
 
-            if not is_host_in_room:
+            # Check if a remote join should be performed.
+            remote_join = not is_host_in_room
+            if not remote_join:
+                # If the host is in the room, but not one of the authorised hosts
+                # for restricted join rules, a remote join must be used.
+                room_version = await self.store.get_room_version(room_id)
+                current_state_ids = await self.store.get_current_state_ids(room_id)
+
+                # Otherwise, check if they should be allowed access via membership in a space.
```

Comment:
Otherwise? We haven't done a check yet?

Reply:
I think this comment got pulled around from copying and pasting. It doesn't seem to make sense anymore.